### PR TITLE
Fail fast in image generation if not enough disk space for customers

### PR DESCRIPTION
**Pull request by nikita-bykov**
Сheck of free disk space on Windows and Ubuntu was added after image generation was completed.

Issue: #752

### CODE REVIEW

File: images/win/scripts/Installers/Validate-DiskSpace.ps1 [hunk 1]
```diff
@@ -0,0 +1,13 @@
+################################################################################
+##  File:  Validate-DiskSpace.ps1
+##  Desc:  Validate free disk space
+################################################################################
+
+$availableFreeSpace = ([IO.DriveInfo]::GetDrives() | Where-Object {$_.Name -eq 'C:\'}).AvailableFreeSpace / 1024 / 1024 / 1024
```
Comment (al-cheb) [inline images/win/scripts/Installers/Validate-DiskSpace.ps1:6]: (Get-PSDrive -Name).Free

File: images/win/scripts/Installers/Validate-DiskSpace.ps1 [hunk 2]
```diff
@@ -0,0 +1,13 @@
+################################################################################
+##  File:  Validate-DiskSpace.ps1
+##  Desc:  Validate free disk space
+################################################################################
+
+$availableFreeSpace = ([IO.DriveInfo]::GetDrives() | Where-Object {$_.Name -eq 'C:\'}).AvailableFreeSpace / 1024 / 1024 / 1024
+$minimumFreeSpaceGB = 15
```
Comment (al-cheb) [inline images/win/scripts/Installers/Validate-DiskSpace.ps1:7]: 15GB

File: images/linux/scripts/installers/validate-disk-space.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+################################################################################
+##  File:  validate-disk-space.sh
+##  Desc:  Validate free disk space
+################################################################################
+
+availableSpace=$(df / | sed 1d | awk '{ print $4 "\t" }')
+minimumFreeSpaceGB=$(( 15*1024*1024 ))
```
Comment (miketimofeev) [inline images/linux/scripts/installers/validate-disk-space.sh:8]: Why 15? Have you seen my comment here?
https://github.com/actions/virtual-environments/issues/752#issuecomment-617378137

Comment (nikita-bykov) [reply images/linux/scripts/installers/validate-disk-space.sh]: Sorry, I missed your comment.
Added additional 8GB

File: images/win/scripts/Installers/Validate-DiskSpace.ps1 [hunk 4]
```diff
@@ -0,0 +1,14 @@
+################################################################################
+##  File:  Validate-DiskSpace.ps1
+##  Desc:  Validate free disk space
+################################################################################
+
+$availableSpaceMB =  [math]::Round((Get-PSDrive -Name C).Free / 1MB)
+$minimumFreeSpaceMB = 15 * 1024
+
+Write-Host "Available disk space: $availableSpaceMB MB"
+if ($availableSpaceMB -le $minimumFreeSpaceGB)
```
Comment (al-cheb) [inline images/win/scripts/Installers/Validate-DiskSpace.ps1:10]: Where do we define $minimumFreeSpaceGB ?

File: images/linux/scripts/installers/validate-disk-space.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+################################################################################
+##  File:  validate-disk-space.sh
+##  Desc:  Validate free disk space
+################################################################################
+
+availableSpaceMB=$(( $(df / | sed 1d | awk '{ print $4 "\t" }')/1024 ))
```
Comment (al-cheb) [inline images/linux/scripts/installers/validate-disk-space.sh:7]: df supports -hm format that returns size in mb.

$ df / -hm
Filesystem     1M-blocks  Used Available Use% Mounted on
rootfs            120019 80502     39518  68% /

Comment (nikita-bykov) [reply images/linux/scripts/installers/validate-disk-space.sh]: Thanks, fixed.

**Pull request merged**